qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -231,11 +254,71 @@ def __init__(self,
             get_cmd=":FUNCtion:IMPedance:RANGe?",
             set_cmd=":FUNCtion:IMPedance:RANGe {}",
             unit='Ohm',
-            vals=Enum(0.1, 1, 10, 100, 300, 1000, 3000, 10000, 30000, 100000),
+            vals=self._imp_range,
             docstring="Selects the impedance measurement range, also turns "
                       "the auto range function OFF."
         )
 
+        self.add_parameter(
+            "dc_bias_enabled",
+            get_cmd=":BIAS:STATe?",
+            set_cmd=":BIAS:STATe {}",
+            vals=Bool(),
+            val_mapping=create_on_off_val_mapping(on_val="1",
+                                                  off_val="0"),
+            docstring="Enables DC bias. DC bias is automatically turned "
+                      "off after recalling the state from memory."
+        )
+
+        self.add_parameter(
+            "dc_bias_voltage_level",
+            get_cmd=":BIAS:VOLTage:LEVel?",
+            set_cmd=":BIAS:VOLTage:LEVel {}",
+            get_parser=float,
+            unit="V",
+            vals=self._dc_bias_v_level_range,
+            docstring="Sets the DC bias voltage. Setting does not "
+                      "implicitly turn the DC bias ON."
+        )
+
+        self.add_parameter(
+            "meas_time_mode",
+            initial_value="MED",
+            vals=Enum("SHOR", "MED", "LONG"),
```

Comment:
i'd recommend to decouple the qcodes driver from the specifics of the VISA command values and use val_mapping here:
```
val_mapping={"short": "SHOR", "medium": "MED", "long": "LONG"}
```
this makes the code that uses the driver more readable. and removes the need for explaining the values in the docstring (i'm referring to the `It can be SHOR (short), MED (medium - default) or LONG.`  sentence :) )